radicale/__init__.py

```diff
@@ -425,6 +433,13 @@ def response(status, headers=(), answer=None):
                     environ, base_prefix, path, user)
             except socket.timeout:
                 return response(*REQUEST_TIMEOUT)
+            except vobject.base.VObjectError as e:
+                return response(*UNPROCESSABLE_ENTITY(e))
```

Comment:
IMO 422 is the correct code, see the explanation in [RFC 4918 section11.2](https://tools.ietf.org/html/rfc4918#section-11.2) – which is exactly describing this case.

I'm not putting it into `do_PUT` because here is where exceptions in requests are handled. And if (for whatever reason) vObjects are parsed in another method, this is also handled .